ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -794,10 +877,14 @@ void processFile(AbstractFile f, DrawableDB.DrawableTransaction tr) {
                     if (FileTypeUtils.hasDrawableMIMEType(f)) {  //supported mimetype => analyzed
                         taskDB.updateFile(DrawableFile.create(f, true, false), tr);
                     } else { //unsupported mimtype => analyzed but shouldn't include
+                        // if mimetype of the file hasnt been ascertained, ingest might not have completed yet.
+                        if (null == f.getMIMEType()) {
+                            this.setTaskCompletionStatus(false);
+                        }
```

Comment:
AbstractFile.getMIMEType will never return null after a call to FileTypeUtils.hasDrawableMIMEType, so the call to setTaskCompletionStatus will never be executed and we do not need the taskCompletionStatus stuff. @bcarrier, apparently this was your idea, what is the behavior you are trying to create?

Reply:
We realized that hasDrawableMIMEType() will determine the mimetype if unknown.  
So we changed the condition to:
if ( null != f.getMIMEType() && FileTypeUtils.hasDrawableMIMEType(f)) {
...
}